File: functions_example/Class1.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace functions_example
{
    public static class FunctionLibrary
    {
        public static decimal sqsum(IEnumerable<decimal> a)
        {
            return a.Sum(x => x * x);
        }

        public static decimal gcd(decimal x, decimal y)
        {
            ulong a = (ulong)x;
            ulong b = (ulong)y;

            while (a != 0 && b != 0)
            {
                if (a > b)
                    a %= b;
                else
                    b %= a;
            }

            return a | b;
        }

        public static IEnumerable<decimal> join_seq(IEnumerable<decimal> a, IEnumerable<decimal> b)
        {
            return a.Concat(b);
        }
    }
}

File: functions_example/VB.NET/Class1.vb
Imports System.ComponentModel
Public NotInheritable Class FunctionLibrary

    <Description("Square sum///sqsum([A:(1,2,3)])")>
    Public Shared Function sqsum(a As IEnumerable(Of Decimal)) As Decimal
        Return a.Sum(Function(x) x * x)
    End Function

    <Description("Greatest common divisor")>
    Public Shared Function gcd(x As Decimal, y As Decimal) As Decimal
        Dim a As ULong = Convert.ToUInt64(x)
        Dim b As ULong = Convert.ToUInt64(y)

        Do While (a <> 0 And b <> 0)
            If a > b Then
                a = a Mod b
            Else
                b = b Mod a
            End If
        Loop

        Return a Or b
    End Function

    <Description("Join ranges")>
    Public Shared Function join_seq(a As IEnumerable(Of Decimal), b As IEnumerable(Of Decimal)) As IEnumerable(Of Decimal)
        Return a.Concat(b)
    End Function
End Class
